### ISSUE TITLE
Multi-Minitor: Windows moved to or spawned on second screen are actually still on the first screen

### ISSUE THREAD

**Issue by idl0r** [awesomeWM/awesome#512]
Based on https://awesome.naquadah.org/bugs/index.php?do=details&task_id=1101
That bug is still present and can be reproduced with e.g. freerdp, rdesktop or mplayer.
Start mplayer, hit MOD+o, check the task bar or switch to a different tag on screen one and you'll see the window disappears on screen 2.

**Comment by blueyed** [comment]
@idl0r 
Are you using Git master?

**Comment by idl0r** [comment]
3.5.6

**Comment by blueyed** [comment]
I think this may be fixed in master.

**Comment by psychon** [comment]
Cannot reproduce with mplayer

**Comment by idl0r** [comment]
Hm, haven't tried master yet but it seems to be broken only with larger videos with mplayer, e.g. 1920x1080 in my case. Smaller sizes aren't affected for some reason.

**Comment by actionless** [comment]
>only with larger videos with mplayer

ie in case when client size is larger than available screen workarea?

**Comment by idl0r** [comment]
The first screen is 1920x1080 and the second is 1680x1050 so in this case is was larger but @work I do have two screens with 1920x1080 and the same happens with rdesktop for example. But I haven't tried mplayer at work. Might be two different issues actually.

Anyway, I also tried master and at the first glance it seems to be fixed but there's also an error when moving (MOD+o):
```
W: awesome: luaA_dofunction:77: error while running function
stack traceback:
	/usr/share/awesome/lib/awful/screen.lua:63: in function 'focus'
	/usr/share/awesome/lib/awful/client.lua:549: in function 'press'
	/usr/share/awesome/lib/awful/key.lua:57: in function </usr/share/awesome/lib/awful/key.lua:57>
error: /usr/share/awesome/lib/awful/screen.lua:63: attempt to index local 'pos' (a nil value)
```

**Comment by psychon** [comment]
That error gets fixed with #516: 5ae5e60f

So I guess this means this problem is somehow fixed in master.

**Comment by idl0r** [comment]
Yes, perfect! Thanks!
